test/should_fail/logic_op.erl

```diff
@@ -1,8 +1,37 @@
 -module(logic_op).
--export([failand/1, failandalso/1]).
+-export([failand/1, failand2/1,
+         failandleft/2, failandleft2/2,
+         failandright/2, failandright2/2,
+         failandalso/1, failandalso2/1]).
 
 -spec failand(boolean()) -> tuple().
 failand(X) -> X and X.
 
+-spec failand2(boolean()) -> tuple().
+failand2(X) ->
+    O = X and X,
+    O.
```

Comment:
Can you explain why do this, rather than just returning `X and X`? I've forgotten why we test things in this way.

Reply:
Just returning `X and X` invokes `type_check_expr` wheareas this invokes `type_check_expr_in`.

Reply:
Ah, right! But vice versa.